src/filter_linux.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifdef __linux__
+
+#include <time.h>
+#include <error.h>
+
+#include "log.h"
+#include "filter.h"
+
+const u64 NS_IN_1_SEC = 1000000000;
+
+int clock_id(bool use_realtime, bool use_unix) {
+    if (use_realtime) {
+        return CLOCK_REALTIME;
+    } else if (use_unix) {
+        return CLOCK_MONOTONIC;
+    } else {
+        Log::error("No clockid set");
+  	return -1;
+    }
+}
+
+HeartBeatFilter::~HeartBeatFilter() {
+    munmap(this->_region_ptr, sizeof(u64));
+    close(this->_fd);  
+}
+
+HeartBeatFilter::HeartBeatFilter(const char* heartbeat_file, u64 interval_ns, bool use_unix_clock, bool use_realtime_clock) {
```

Comment:
Can we unify the naming of the variable holding threshold to accept the profile? It's `interval_ns` in some files, and `delay_ns` in some others. I'd use `threshold_ns` for all of them, what do you think?